### PR TITLE
event: add on hold info

### PR DESCRIPTION
**Pull request by copiltembel**
This PR:

Adds the call.on_hold information to the event dictionary

### CODE REVIEW

File: src/event.c [hunk 1]
```diff
@@ -208,13 +208,17 @@ int event_encode_dict(struct odict *od, struct ua *ua, enum ua_event ev,
 			err |= odict_entry_add(od, "diverteruri", ODICT_STRING,
 					       call_diverteruri(call));
 
+		err |= odict_entry_add(od, "onhold", ODICT_BOOL,
+				call_is_onhold(call));
+
```
Comment (cspiel1) [inline src/event.c:213]: This is only the local on-hold flag. The application should know that it put the call on hold locally. Is this worth to be put to the event?
Should we add it only if `call_is_onhold()`?

File: src/event.c [hunk 2]
```diff
@@ -208,13 +208,17 @@ int event_encode_dict(struct odict *od, struct ua *ua, enum ua_event ev,
...
 		const char *user_data = call_user_data(call);
 		if (user_data) {
 			err |= odict_entry_add(od, "userdata", ODICT_STRING,
 				user_data);
-			if (err)
-				goto out;
 		}
+
+		if (err)
+			goto out;
```
Comment (cspiel1) [inline src/event.c:221]: I think this should be at least a separate commit. Normally we make separate PRs, for different fixes/features.

**Comment by juha-h** [comment]
Christian Spielberger writes:

> This is only the local on-hold flag. The application should know that it put the call on hold locally. Is this worth to be put to the event?
> Should we add it only if `call_is_onhold()`?

In my Android app, call has two properties: call.onhold and call.held.
So both are needed.

File: src/event.c (hunk 2 above)
Comment (copiltembel) [reply src/event.c]: OK, I've moved this change to a separate PR: https://github.com/baresip/baresip/pull/1992

File: src/event.c (hunk 1 above)
Comment (copiltembel) [reply src/event.c:213]: For me it is important to know if a call has been put on hold, regardless if it was triggered by the local client or the remote party. I am using the mqtt module to get updates about the call states and HOLD is something I am interested in.

If the remote party sets the call to HOLD, the call_is_onhold is not updated?

**Comment by sreimers** [comment]
You can rebase/merge main to fix clang ci issues.

**Comment by juha-h** [comment]
copiltembel writes:

> For me it is important to know if a call has been put on hold, regardless if it was triggered by the local client or the remote party. I am using the mqtt module to get updates about the call states and HOLD is something I am interested in.
> 
> If the remote party sets the call to HOLD, the call_is_onhold is not
> updated?

Haven't checked, but a call is on hold when local party has put the call
on hold and the call is held when remote party has put the call on hold.

When call is put on hold by one party of the call, re-INVITE has 

a=sendonly

and 200 OK has

a=recvonly

If also the other party puts the call on hold, 200 OK has

a=inactive

So two separate checks would be needed: is call on hold (i.e.. put on
hold by the local party) and is call held (i.e., put on hold by the
remote party).

**Comment by copiltembel** [comment]
> So two separate checks would be needed: is call on hold (i.e.. put on
> hold by the local party) and is call held (i.e., put on hold by the
> remote party).

That makes sense! So we need a second field on the call that keeps track of the "is call held" status.

**Comment by cspiel1** [comment]
If a re-INVITE has `a=sendonly`, then "the remote put us on-hold" is only an interpretation. This could be done by the application. The audio directions are already encoded to the event data.
